core/templates/dev/head/components/RuleEditorDirective.js

```diff
@@ -143,8 +143,13 @@ oppia.directive('ruleEditor', ['$log', function($log) {
         });
 
         $scope.onSelectNewRuleType = function(newRuleType) {
+          var oldRuleInputs = $scope.rule.inputs || {};
```

Comment:
For safety (and to protect against future changes by developers below), it might be prudent to do angular.copy here and in the line below.